src/tensorlake/applications/image.py

```diff
@@ -23,7 +23,7 @@
 class ImageInformation:
     image: Image
     # Functions that are using the image.
-    functions: List[Function]
+    functions: set[Function]
```

Comment:
is this a breaking change? should i change this back?

Reply:
This won't solve the problem afaik, because we need to dedup images, not Functions in Dict[Image, ImageInformation] returned by image_infos() and they are already deduped, see line 35.

Besides that Functions are not intended to be stored in sets, they don't have == operator or hash implemented.

What needs to be done on image builder side afaik is to not build image per function but instead build image once and then associate it with every function that uses it.